mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2919,24 +2919,28 @@ pub fn burn_address<T: UtxoCommonOps + SwapOps>(coin: &T) -> Result<Address, Str
 
 /// Hash message for signature using Bitcoin's message signing format.
 /// sha256(sha256(PREFIX_LENGTH + PREFIX + MESSAGE_LENGTH + MESSAGE))
-pub fn sign_message_hash(coin: &UtxoCoinFields, message: &str) -> Option<[u8; 32]> {
-    let message_prefix = coin.conf.sign_message_prefix.clone()?;
+pub fn sign_message_hash(sign_message_prefix: &str, message: &str) -> [u8; 32] {
     let mut stream = Stream::new();
-    let prefix_len = CompactInteger::from(message_prefix.len());
+    let prefix_len = CompactInteger::from(sign_message_prefix.len());
     prefix_len.serialize(&mut stream);
-    stream.append_slice(message_prefix.as_bytes());
+    stream.append_slice(sign_message_prefix.as_bytes());
     let msg_len = CompactInteger::from(message.len());
     msg_len.serialize(&mut stream);
     stream.append_slice(message.as_bytes());
-    Some(dhash256(&stream.out()).take())
+    dhash256(&stream.out()).take()
 }
```

Comment:
I guess this shouldn't be part of utxo_common anymore and be part of mm2_bitcoin crate instead. Not urgent in this PR and can be done later. You can add a todo if you won't move it.

Reply:
[6521246](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2499/commits/65212468416c1576e9a198250254364410a756f8)